memgpt/models/pydantic_models.py

```diff
@@ -100,6 +100,7 @@ class SourceModel(SQLModel, table=True):
     embedding_config: Optional[EmbeddingConfigModel] = Field(
         None, sa_column=Column(JSON), description="The embedding configuration used by the passage."
     )
+    source_metadata: Optional[dict] = Field(None, sa_column=Column(JSON), description="Metadata associated with the source.")
```

Comment:
@sarahwooders I'm using the attribute `source_metadata` instead of `metadata` because `metadata` is reserved for any model that inherits from `SQLModel`